### PR TITLE
fix redirect handling

### PR DESCRIPTION
**Pull request by hpdev**

### CODE REVIEW

File: lib/dumbo/task/base.rb [hunk 1]
```diff
@@ -31,8 +31,12 @@ def run!(overlord)
         @id = MultiJson.load(response.body)["task"]
       end
 
-      def finished?
-        uri = URI(@overlord + "/#{@id}/status")
+      def finished?(redirect)
+        if redirect
+          uri = URI(redirect)
+        else 
+          URI(@overlord + "/#{@id}/status")
```
Comment (danchel) [inline lib/dumbo/task/base.rb]: I think this should be: 
```
uri = URI(@overlord + "/#{@id}/status")
```

File: lib/dumbo/task/base.rb [hunk 2]
```diff
@@ -31,8 +31,12 @@ def run!(overlord)
         @id = MultiJson.load(response.body)["task"]
       end
 
-      def finished?
-        uri = URI(@overlord + "/#{@id}/status")
+      def finished?(redirect)
```
Comment (danchel) [inline lib/dumbo/task/base.rb]: If its called without the redirect param somewhere (which is most likely the case) it has to be

```
      def finished?(redirect = nil)
```

File: lib/dumbo/task/base.rb (hunk 1 above)
Comment (hpdev) [reply lib/dumbo/task/base.rb]: Done

File: lib/dumbo/task/base.rb (hunk 2 above)
Comment (hpdev) [reply lib/dumbo/task/base.rb]: Done

**Comment by adwinsky** [comment]
👍

**Pull request merged**